### PR TITLE
doc/dev: add health-reports.rst

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
https://ceph--42582.org.readthedocs.build/en/42582/dev/health-reports/

File: doc/dev/health-reports.rst [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+==================
+
+In general, there are two channels to retrieve the health reports:
+
+ceph (CLI)
+   which send ``health`` mon command for retrieving the health status of the cluster
+mgr module
+   which calls ``mgr.get('health')`` for the same report in the form of a JSON encoded string
+
+Following diagrams outline the involved parties and how the interact when the clients
```
Comment (rzarzynski) [inline doc/dev/health-reports.rst:16]: s/Following/The following/, I guess (but I'm not a native speaker).

**Review by rzarzynski — APPROVED** [review]
Generally LGTM; just some minors / nits.

File: doc/dev/health-reports.rst [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+==============
+
+
+How to Get Reports
+==================
+
+In general, there are two channels to retrieve the health reports:
+
+ceph (CLI)
+   which send ``health`` mon command for retrieving the health status of the cluster
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply doc/dev/health-reports.rst:12]: fixed.

File: doc/dev/health-reports.rst [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+- MgrMonitor
+- MgrStatMonitor
+- MonmapMonitor
+- OSDMonitor
+
+When persisting the pending changes in their own domain, each of them identifies the
+health related issues and store them into the monstore with the prefix of ``health``
+using the same transaction. For instance, ``OSDMonitor`` checks a pending new osdmap
+for possible issues, like down OSDs and missing scrub flag in a pool, and then stores
+the encoded form of the health reports along with the new osdmap. These reports are later loaded and decoded, so they can be collected on demand. When it comes to ``MDSMonitor``,
```
[the remark this answers is not in the document]
Comment (tchaikov) [reply doc/dev/health-reports.rst]: done.

File: doc/dev/health-reports.rst (hunk 1 above)
Comment (tchaikov) [reply doc/dev/health-reports.rst]: fixed.

**Pull request merged**